File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        
        if len(s) > len(t): return False
        if not len(s): return True
        
        pt1, pt2 = 0, 0
        
        while pt2 < len(t):
            if s[pt1] == t[pt2]:
                pt1 += 1
                if pt1 == len(s): return True
            pt2 += 1
        
        
        return pt1 == len(s)
        

File: 0392-is-subsequence/0392-is-subsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        
        if (t.Length < s.Length) return false;
        int i = 0;
        int j = 0;

        while (i < s.Length && j < t.Length) {
            if (s[i] == t[j]) {
                i++; j++;
            } else j++;
        }

        if (i == s.Length) return true;
        return false;



    }
}